pkg/parse/root.go

```diff
@@ -141,7 +146,9 @@ func (p *root) parseSource(_ context.Context, state sourceState) ([]ast.FileObje
 	options = OptionsForScope(options, p.scope)
 
 	if p.sourceFormat == filesystem.SourceFormatUnstructured {
-		options.Visitors = append(options.Visitors, p.addImplicitNamespaces)
+		if p.namespaceStrategy == configsync.NamespaceStrategyImplicit {
+			options.Visitors = append(options.Visitors, p.addImplicitNamespaces)
```

Comment:
Does this mean that the "explicit" namespace mode isn't supported in hierarchical mode?